newtest/reloader_test.go

```diff
@@ -0,0 +1,255 @@
+package newtest
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/mesosphere/dkp-test-framework/pkg/addons"
+	"github.com/mesosphere/dkp-test-framework/pkg/cluster"
+	"github.com/mesosphere/dkp-test-framework/pkg/cluster/kind"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+func deployKubeaddonsController(cluster cluster.Cluster) error {
```

Comment:
Out of curiosity (since I tried to anticipate this need) did you not see, or not like the available helper function https://github.com/mesosphere/dkp-test-framework/blob/main/pkg/addons/kubeaddons.go#L51?

Reply:
That's part of my feedback. Honestly, I don't believe kubeaddons should be forced onto users of this framework. We might possibly enable it through an additional Go Module but kubeaddons hauls in so many deps and requires consumers to `replace` deps that I'd rather not like to include it anywhere in my code.

Reply:
Not feeling like we're able to use libraries that are intended to be used as libraries seems like a huge fundamental problem to me. I'll think on this for a bit :thinking:

Reply:
I'm also not sure from a scope level of dkp-test-framework whether it should expose anything kubeaddons at all. That's something we should discuss.

Reply:
Kubeaddons does bring with it a lot of dependencies, which of course if a good thing to avoid if possible. One thing that we did with konvoy and yakcl is to have a separate API (and indeed client) module that can be imported with minimal dependencies. This might be a good idea for kubeaddons to reduce the amount of transitive dependencies being imported into consumers of this test framework. Thoughts?

Reply:
I agree with that @jimmidyson said above and that's actually what I meant to say when I said
> We might possibly enable it through an additional Go Module

Reply:
I'm in favor of splitting up the kubeaddons lib into multiple go modules for the goal of being able to use it in the testing framework, as I feel it can be invaluable for test cases.

I've created https://jira.d2iq.com/browse/D2IQ-73374 as a follow up.

In the meantime, would you be amenable to switching to the functionality that exists for the time being? Or what are your thoughts on that?